kvbc/src/categorization/block_merkle_category.cpp

```diff
@@ -541,11 +544,17 @@ std::pair<SetOfKeyValuePairs, KeysVector> BlockMerkleCategory::rewriteAlreadyPru
   return std::make_pair(merkle_blocks_to_rewrite, merkle_blocks_to_delete);
 }
 
-void BlockMerkleCategory::deleteGenesisBlock(BlockId block_id, const BlockMerkleOutput& out, NativeWriteBatch& batch) {
+size_t BlockMerkleCategory::deleteGenesisBlock(BlockId block_id,
+                                               const BlockMerkleOutput& out,
+                                               NativeWriteBatch& batch) {
   auto [hashed_keys, latest_versions] = getLatestVersions(out);
   auto overwritten_active_keys_from_pruned_blocks = findActiveKeysFromPrunedBlocks(hashed_keys);
+  size_t num_of_deletes = 0;
+  for (auto& kv : overwritten_active_keys_from_pruned_blocks) {
+    num_of_deletes += kv.second.size();
```

Comment:
Are we counting key deletes from disk only or stale data too?

Reply:
stale only

Reply:
@andrewjstone Could you please also have a look at the block merkle category?

Reply:
Looks fine to me. It's somewhat strange that we have 3 ways to return data, but 🤷